### DISCUSSION TITLE
Lubuntu 18.04 and novelWriter updates

### DISCUSSION THREAD

**Discussion by HeyMyian** [vkbo/novelWriter#1296] [Help] [answered]
Hi! I'm not really looking to upgrade Lubuntu from 18.04, because I'm already on an old machine with weak hardware. As I understand, Python 3.6 is the last version officially supported in 18.04, and new releases of novelWriter do not support Python 3.6 anymore.

I'm looking to manually upgrade Python to 3.8 and while I have thankfully been hit over the head with the information
> DON'T EVER CHANGE DEFAULT PYTHON!!!

I'm wondering if this: https://askubuntu.com/a/1197712 would be a solution that novelWriter 2.0.2 is compatible with -- I'm tech savy, but still very much a layperson when it comes to anything Linux. I'm imagining with this, I would be able to run novelWriter 2.0.2 with alternate Python version 3.8 from the terminal? But I'm really unsure to try.

Is there anything else you can suggest, or is it a lost cause? novelWriter is the main use of this Lubuntu 18.04 machine.

**Comment by johnblommers** [comment, marked as answer]
Try the AppImage version of novelWriter. AppImages bundle in their dependencies so you should be all right on Ubuntu 18.04 which supports the AppImage format. I just installed Ubuntu 18.04.6 LTS in a live Boxes VM with just 2 GB RAM allocated, downloaded 2.02 AppImage and created a new project with the Wizard. I did not need to fuss with installing anything.

Let me applaud you for keeping your old iron working by using Linux.

**Reply by HeyMyian** [reply]
Charmed. Thank you so much for the hint and the quick response. The AppImage works in Lubuntu 18.04. I also created a `.desktop` file for the AppImage and set `.nwx` files to open with it on default.

My GUI and syntax themes as well as some (don't ask) settings were reset, but that might have been because of user error; I don't recall. Luckily I had backups. Since the icon theme isn't in the general novelWriter settings anymore, I manually added that line to my theme config file.

omigod, first look at novelWriter 2. It looks so slick! I'm not one for change, but I'm genuinely excited for this. :blush:

**Reply by johnblommers** [reply]
Good to hear it. You are correct that novelWriter is a helluva good looking application. Given that a writer spends a lot of time with their writing tool this really helps make writing more pleasant in novelWriter.

**Reply by vkbo** [reply]
Yeah, sorry about having to drop 3.6. I also need to drop 3.7 when I can as both these versions block me from doing certain things I want to do with the source code. Dropping 3.7 will make it possible to also drop the lxml dependency, and instead use the standard library xml package. Thanks John for pointing out AppImage. It should be the universal solution on Linux. We're also working on Flatpak.

I'm glad to hear you like the new design. It took me nearly a year to make the 2.0 version with all the features. There's a lot of refactoring under the hood as well. 😄
